src/framework/ui/view/interactiveprovider.cpp

```diff
@@ -632,8 +632,11 @@ void InteractiveProvider::onOpen(const QVariant& type, const QVariant& objectId,
     }
 
     if (ContainerType::PrimaryPage == containerType) {
-        m_stack.clear();
-        m_stack.push(m_openingObject);
+        if (m_stack.empty()) {
+            m_stack.push(m_openingObject);
+        } else {
+            m_stack[0] = m_openingObject;
```

Comment:
Could you please explain the logic. Maybe even better to add a comment to code. It is not very clear what is happening.